test/show.jl

```diff
@@ -1181,3 +1181,6 @@ end
     @test repr("text/plain", context=:compact=>true) == "\"text/plain\""
     @test repr(MIME("text/plain"), context=:compact=>true) == "MIME type text/plain"
 end
+
+# issue #25857
+@test repr([(1,),(1,2),(1,2,3)]) == "Tuple{Int64,Vararg{Int64,N} where N}[(1,), (1, 2), (1, 2, 3)]"
```

Comment:
I wonder if `Int64` should be changed to `$Int` to pass on 32 bits machines. Also, there is a "typeinfo" `@testset` already, you could insert this test there (at line 1124).